utils/modules/edge_utils.py

```diff
@@ -81,3 +130,135 @@ def faces_angle(normals, edges, pols):
+    bm.free()
+    vals = edges_normal(vertices, normals, edges, faces)
+
+    return vals
+
+def edges_normal(vertices, normals, edges, pols):
+    '''Average of normals of the faces that share the edge (only the two first)'''
+    e_sorted = [sorted(e) for e in edges]
+    ad_faces = [[] for e in edges]
+    for idp, pol in enumerate(pols):
+        for edge in zip(pol, pol[1:] + [pol[0]]):
+            e_s = sorted(edge)
+            if e_s in e_sorted:
+                idx = e_sorted.index(e_s)
+                ad_faces[idx].append(idp)
+    result = []
+
+    for edg_f, edg in zip(ad_faces, edges):
+        if len(edg_f) > 1:
+            edge_normal = ((Vector(normals[edg_f[0]])+Vector(normals[edg_f[1]]))/2).normalized()
+        elif len(edg_f) == 1:
+            edge_normal = Vector(normals[edg_f[0]])
+        else:
+            rot_mat = Matrix.Rotation(pi/2, 4, "Z")
+            direc = (Vector(vertices[edg[1]]) - Vector(vertices[edg[0]])).normalized()
+            edge_normal = direc @ rot_mat
+        result.append(tuple(edge_normal))
+
+    return result
+
+
+def edges_vertices(vertices, edges):
+    '''Explode edges'''
+    verts = [[vertices[c] for c in e] for e in edges]
+    eds = [[[0, 1]] for e in edges]
+    vals = [verts, eds]
+    return vals
+
+def edge_is_filter(vertices, edges, faces, mode):
+    '''acces to mesh_filter to get differnt bmesh edges filters'''
+    mode = mode.split(' ')[1]
+    bm = bmesh_from_pydata(vertices, edges, faces, normal_update=True)
+    good, bad, mask = Edges.process(bm, mode, edges)
+    bm.free()
+    return mask, good, bad
+
+
+def edges_shell_factor(vertices, edges, faces):
+    '''Average of vertex shell_factor'''
+    v_shell = vertex_shell_factor(vertices, edges, faces)
+    vals = [(v_shell[e[0]] + v_shell[e[1]])/2 for e in edges]
+    return vals
+
+def edges_center(vertices, edges):
+    vals = [tuple((Vector(vertices[e[0]])+Vector(vertices[e[1]]))/2) for e in edges]
+    return vals
+
+def edges_origin(vertices, edges):
+    vals = [vertices[e[0]] for e in edges]
+    return vals
+
+def edges_end(vertices, edges):
+    vals = [vertices[e[1]] for e in edges]
+    return vals
+
+def edges_inverted(vertices, edges):
+    vals = [[e[1], e[0]] for e in edges]
+    return vals
+
+def edge_vertex(vertices, edges, origin):
+    if origin == 'Center':
+        center = [(Vector(vertices[e[0]])+Vector(vertices[e[1]]))/2 for e in edges]
+    elif origin == 'First':
+        center = [Vector(vertices[e[0]]) for e in edges]
+    else:
+        center = [Vector(vertices[e[1]]) for e in edges]
+    return center
+
+def edges_matrix(vertices, edges, orientation):
+    '''Matrix aligned with edge.'''
+    origin, track, up = orientation
+    normal = edges_direction(vertices, edges, out_numpy=False)
+    normal_v = [Vector(n) for n in normal]
+    center = edge_vertex(vertices, edges, origin)
+    vals = matrix_normal([center, normal_v], track, up)
+    return vals
```

Comment:
I think if function are in utils it means that it can be used in another nodes by other developers. I would appreciate if functions have more explicit documentation. What are expecting in input and what will be in output like this:
```Python
"""
vertices: list of tuple(flaot, float, float)
edges: list of tuple(int, int)
orientation: tuple(origin, track, up) where origin is list of tuple(float, float, float) and so on
return: list of Matrixes
"""
```

Reply:
Ok, I will expand my code documentation skills